src/Symfony/Bridge/Doctrine/Form/ChoiceList/EntityChoiceList.php

```diff
@@ -126,6 +129,8 @@ public function __construct(ObjectManager $manager, $class, $labelPath = null, E
         }
 
         parent::__construct($entities, $labelPath, $preferredEntities, $groupPath, null, $propertyAccessor);
+
+        trigger_error('The '.__CLASS__.' class is deprecated since version 2.7 and will be removed in 3.0. Use Symfony\Bridge\Doctrine\Form\ChoiceList\DoctrineChoiceLoader instead.', E_USER_DEPRECATED);
```

Comment:
https://github.com/symfony/symfony/blob/5188cfecc58c78bda18cd7081ea86ea69f0d596c/src/Symfony/Component/Validator/Validator/LegacyValidator.php#L14

this way the trigger happens the first time the class is autoloaded